tests/test_client.py

```diff
@@ -0,0 +1,56 @@
+import unittest
+import requests
+import aiohttp
+
+from unittest.mock import MagicMock, AsyncMock
+from sdc_client import client
+from sdc_client.base_api_client import UnauthorizedException
+from conftest import MockResponse, MockAsyncResponse, StreamSetsMock
+
+
+class TestGetJMX(unittest.TestCase):
+    def test_get_jmx_success(self):
+        requests.Session.get = MagicMock(return_value=MockResponse(
+            _text='data',
+            status_code=200
+        ))
+        result = client.get_jmx(StreamSetsMock(), 'query_params_1')
+        self.assertEqual(result, {'json': 'data'})
+
+    def test_get_jmx_unauthorized(self):
+        requests.Session.get = MagicMock(return_value=MockResponse(
+            _text='data',
+            status_code=401,
+        ))
+        try:
+            result = client.get_jmx(StreamSetsMock(), 'query_params_1')
+        except Exception as e:
+            self.assertIsInstance(e, UnauthorizedException)
+        else:
+            raise Exception("Code must raise error for this case")
```

Comment:
You can use assertRaises() from unittest if an exception expected.
```python
with self.assertRaises(UnauthorizedException):
    client.get_jmx(StreamSetsMock(), 'query_params_1')
```

Reply:
I looked for it, thanks